ecosystem/sep-0031.md

```diff
@@ -442,14 +441,23 @@ Name | Type | Description
 
 ##### Success (201 Created)
 
-This is the successful case where a Receiving Anchor confirms that they can fulfill this payment as described. The response body should be a JSON object with the following values
+This is the successful case where a Receiving Anchor either confirms that they can fulfill this payment as described or that confirmation is pending.
+
+Anchors should check the `status` response attribute to determine whether or not the Receiving Anchor is ready to receive the payment. 
+
+If `status` is `pending_sender`, `stellar_account_id`, `stellar_memo`, & `stellar_memo_type` should be present. 
+
+If `status` is `pending_receiver`, the Receiving Anchor is processing the information sent and determining if the transaction can proceed. In this case, `stellar_account_id`, `stellar_memo`, & `stellar_memo_type` will not be present, and the Sending Anchor should monitor the transaction's status until it moves to `pending_sender` or `error`. 
+
+Sending Anchors can monitor the transaction's status by registering a callback URL via [`PUT /transactions/:id/callback`](#put-transaction-callback) or by polling [`GET /transactions/:id`](#get-transaction). When the transaction status moves to `pending_sender`, the Stellar account & memo fields should be populated.
 
 Name | Type | Description
 -----|------|------------
 `id` | string | The persistent identifier to check the status of this payment transaction.
-`stellar_account_id` | string | The Stellar account to send payment to.
-`stellar_memo_type` | string | The type of memo to attach to the Stellar payment (`text`, `hash`, or `id`).
-`stellar_memo` | string | The memo to attach to the Stellar payment.
+`stellar_account_id` | string | (optional) The Stellar account to send payment to.
+`stellar_memo_type` | string | (optional) The type of memo to attach to the Stellar payment (`text`, `hash`, or `id`).
+`stellar_memo` | string | (optional) The memo to attach to the Stellar payment.
+`status` | string | (optional) The initial status of the transaction, either `pending_sender` or `pending_receiver`.
```

Comment:
It's unfortunate that we have four optional parameters that are bound together, i.e. not individually optional, they're only optional together.

A status that isn't present is interesting, would it make sense do you think to add a status to represent this moment? It would mean the state machine always has a status, rather than this unknown status period.

Reply:
I added `status` as optional because existing implementations don't return it, not because there isn't a status value that represents the moment. `pending_receiver` would represent the moment if the address & memo fields are not returned, otherwise `pending_sender` would be appropriate.

If we go with @marcelosalloum's suggestion to return the same response body as the `GET /transactions/:id`, then `status` would be required.

So I think we just need to decide how big of a breaking change we want to make. Making previously required response attributes optional is less impactful to existing implementations than making previously absent fields required.

Reply:
Ah, interesting. I wonder if breaking this change into two changes in the changelog will help people understand it clearer. i.e. add status as optional, then make other changes. It doesn't improve the resulting situation though.

Reply:
🤔 I do think there are two breaking changes here

1. Make `stellar_account_id`,  `stellar_memo`, & `stellar_memo_type` optional in `POST /transactions` & `GET /transactions/:id` responses
2. Replace current `POST /transactions` response body with the response body of `GET /transactions/:id`

It just pains me to bump the major version twice. Making 1 breaking change per major version bump makes things clearer, but I see other pieces of software make many breaking changes per major version bump.

Reply:
Ah yeah, we don't want to bump it twice. Breaking changes in software is one thing, in standards it creates more work, as how do clients interoperate with all of them.

Reply:
Ok.. then @marcelosalloum I think we should just go with breaking change 1 and not do 2. It would make the API cleaner and more RESTful but its not worth making current implementations out of date.